platforms.go

```diff
@@ -121,7 +121,8 @@ import (
 )
 
 var (
-	specifierRe = regexp.MustCompile(`^[A-Za-z0-9_-]+$`)
+	specifierRe        = regexp.MustCompile(`^[()A-Za-z0-9_.-]+$`)
```

Comment:
Is this regex too permissive?

For fun, I tried some options in ``, and values like below do not return an error;

- `linux)()---()..../arm/foo`
- `../arm/foo`

(the regex before this was already a bit sketchy though, as it also allowed `--` as prefix), I think the intent here was to (at least for the first component) allow the existing set `[A-Za-z0-9_-]+`, optionally followed by content surrounded by `()`.

Perhaps this would work (tests pass with it at least);

```suggestion
	specifierRe        = regexp.MustCompile(`^[A-Za-z0-9_-]+(\([A-Za-z0-9_.-]*\))?$`)
```

Not sure how strict we want the `<OSVersion>` to be; if we only expect the versions as used by windows (one or more dot-separated digits), then it could be;

```suggestion
	specifierRe        = regexp.MustCompile(`^[A-Za-z0-9_-]+(\(\d*(.\d+)*\))?$`)
```


As we're already using this regex unconditionally, we could use capture-groups to also return the matches. That way we wouldn't need to do the splitting again in `normalizeOS()`;


```suggestion
	specifierRe = regexp.MustCompile(`^([A-Za-z0-9_-]+)(?:\((\d*(.\d+)*)\))?$`)
```

Reply:
> Not sure how strict we want the `<OSVersion>` to be; if we only expect the versions as used by windows (one or more dot-separated digits), then it could be;
>
Not sure we want to enforce this keeping only windows in mind as the format for different OS-es could be different. I think we should take in whatever is honored and have the platform matcher or smth else which actually reads or uses this field to error out if this is not valid.